src/field/FieldEquation.i.hh

```diff
@@ -0,0 +1,73 @@
+//---------------------------------*-CUDA-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file FieldEquation.i.cuh
+//---------------------------------------------------------------------------//
+#include "MagField.hh"
+#include <cmath>
+#include "base/Constants.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with a magnetic field.
+ */
+CELER_FUNCTION
+FieldEquation::FieldEquation(MagField& field) : field_(field)
+{
+    // set the default charge and the lorentz_cof with -eplus
+    set_charge(-1.0);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Set charge for this track and recalcuate lorentz_cof
+ */
+
+CELER_FUNCTION void FieldEquation::set_charge(real_type charge)
+{
+    charge_ = charge;
+    coeffi_ = charge_ * constants::c_light * 1.0e-8;
```

Comment:
This additional conversion factor will go away in the next extension (using celeritas units everywhere in field codes).  If you don't mind, I like to leave like this (instead of introducing a temporary function, static constexpr real_type unit_conversion().

Reply:
Ah I see. The policy we've adopted to far is for code merged into Celeritas be consistent (with the rest of the code) and correct (as far as can be initially determined), though not necessarily complete. If the units are a problem for now, could we defer the parts that use those units until the next PR?